neqo-qpack/src/header_block.rs

```diff
@@ -0,0 +1,971 @@
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+use crate::qpack_helper::{to_string, ReceiverBufferWrapper};
+use crate::qpack_send_buf::QPData;
+use crate::table::HeaderTable;
+use crate::Header;
+use crate::{Error, Res};
+use crate::{Prefix, NO_PREFIX};
+use neqo_common::qtrace;
+use std::ops::Deref;
+
+const BASE_PREFIX_POSITIVE: Prefix = Prefix {
```

Comment:
Given the way that prefixes are fragile, I wonder if a macro might not be a bad idea here.  You can pack a static assertion about the relationship between prefix and length into that using the `static_assertions` crate.